src/redux/news/news.sagas.js

```diff
@@ -29,16 +29,63 @@ function* handleNewsLoad() {
                   primary{
                     medium
                   }
+                  additional{
+                    medium
+                  }
                 }
+                video
                }
              }`
     );
     yield put(setNews(news.data.getAllNews));
+    yield put(setLoading(news.loading));
+  } catch (error) {
+    console.log(error);
+  }
+}
+
+function* handleArticleLoad({ payload }) {
+  try {
+    const article = yield call(
+      getItems,
+      `query{
+        getNewsById(id:"${payload}"){
+          _id
+          title{
+            value
+          }
+          text{
+            value
+          }
+          images{
+            primary{
+              medium
+            }
+            additional{
+              medium
+            }
+          }
+          video
+          author{
+            name{
+              value
+            }
+            image{
+              small
+            }
+          }
+          date
+        }
+      }`
+    );
+    yield put(setArticle(article.data.getNewsById));
+    yield put(setLoading(article.loading));
```

Comment:
Where do you take `article.loading`? Does the response contain any loading stuff? I thought loading should be taken from redux state

Reply:
Saga response has data with data and loading indicator that tells us is it true or false